pkg/events/parsers/data_parsers.go

```diff
@@ -760,156 +760,71 @@ func ParsePtraceRequestArgument(request uint64) (string, error) {
-}
-
-var socketcallCallStringMap = map[SocketcallCallArgument]string{
-	SYS_SOCKET:      "SYS_SOCKET",
-	SYS_BIND:        "SYS_BIND",
-	SYS_CONNECT:     "SYS_CONNECT",
-	SYS_LISTEN:      "SYS_LISTEN",
-	SYS_ACCEPT:      "SYS_ACCEPT",
-	SYS_GETSOCKNAME: "SYS_GETSOCKNAME",
-	SYS_GETPEERNAME: "SYS_GETPEERNAME",
-	SYS_SOCKETPAIR:  "SYS_SOCKETPAIR",
-	SYS_SEND:        "SYS_SEND",
-	SYS_RECV:        "SYS_RECV",
-	SYS_SENDTO:      "SYS_SENDTO",
-	SYS_RECVFROM:    "SYS_RECVFROM",
-	SYS_SHUTDOWN:    "SYS_SHUTDOWN",
-	SYS_SETSOCKOPT:  "SYS_SETSOCKOPT",
-	SYS_GETSOCKOPT:  "SYS_GETSOCKOPT",
-	SYS_SENDMSG:     "SYS_SENDMSG",
-	SYS_RECVMSG:     "SYS_RECVMSG",
-	SYS_ACCEPT4:     "SYS_ACCEPT4",
-	SYS_RECVMMSG:    "SYS_RECVMMSG",
-	SYS_SENDMMSG:    "SYS_SENDMMSG",
-}
-
-func (s SocketcallCallArgument) String() string {
-	var res string
-
-	if sdName, ok := socketcallCallStringMap[s]; ok {
-		res = sdName
-	} else {
-		res = strconv.Itoa(int(s))
-	}
-
-	return res
+var sysNetValues = []SystemFunctionArgument{
+	SYS_SOCKET,
+	SYS_BIND,
+	SYS_CONNECT,
+	SYS_LISTEN,
+	SYS_ACCEPT,
+	SYS_GETSOCKNAME,
+	SYS_GETPEERNAME,
+	SYS_SOCKETPAIR,
+	SYS_SEND,
+	SYS_RECV,
+	SYS_SENDTO,
+	SYS_RECVFROM,
+	SYS_SHUTDOWN,
+	SYS_SETSOCKOPT,
+	SYS_GETSOCKOPT,
+	SYS_SENDMSG,
+	SYS_RECVMSG,
+	SYS_ACCEPT4,
+	SYS_RECVMMSG,
+	SYS_SENDMMSG,
 }
 
-var socketcallCallMap = map[uint64]SocketcallCallArgument{
-	SYS_SOCKET.Value():      SYS_SOCKET,
-	SYS_BIND.Value():        SYS_BIND,
-	SYS_CONNECT.Value():     SYS_CONNECT,
-	SYS_LISTEN.Value():      SYS_LISTEN,
-	SYS_ACCEPT.Value():      SYS_ACCEPT,
-	SYS_GETSOCKNAME.Value(): SYS_GETSOCKNAME,
-	SYS_GETPEERNAME.Value(): SYS_GETPEERNAME,
-	SYS_SOCKETPAIR.Value():  SYS_SOCKETPAIR,
-	SYS_SEND.Value():        SYS_SEND,
-	SYS_RECV.Value():        SYS_RECV,
-	SYS_SENDTO.Value():      SYS_SENDTO,
-	SYS_RECVFROM.Value():    SYS_RECVFROM,
-	SYS_SHUTDOWN.Value():    SYS_SHUTDOWN,
-	SYS_SETSOCKOPT.Value():  SYS_SETSOCKOPT,
-	SYS_GETSOCKOPT.Value():  SYS_GETSOCKOPT,
-	SYS_SENDMSG.Value():     SYS_SENDMSG,
-	SYS_RECVMSG.Value():     SYS_RECVMSG,
-	SYS_ACCEPT4.Value():     SYS_ACCEPT4,
-	SYS_RECVMMSG.Value():    SYS_RECVMMSG,
-	SYS_SENDMMSG.Value():    SYS_SENDMMSG,
-}
+var (
+	SYS_NET_FIRST = SYS_SOCKET.Value()
+	SYS_NET_LAST  = SYS_SENDMMSG.Value()
+)
 
 // ParseSocketcallCall parses the `call` argument of the `socketcall` syscall
 // http://man7.org/linux/man-pages/man2/socketcall.2.html
 // https://elixir.bootlin.com/linux/latest/source/include/uapi/linux/net.h
-func ParseSocketcallCall(rawValue uint64) (SocketcallCallArgument, error) {
-	if v, ok := socketcallCallMap[rawValue]; ok {
-		return v, nil
+func ParseSocketcallCall(call uint64) (string, error) {
+	if call > SYS_NET_LAST {
+		return "", fmt.Errorf("not a valid socketcall call value: %d", call)
 	}
 
-	return 0, fmt.Errorf("not a valid socketcall call value: %d", rawValue)
+	idx := int(call - SYS_NET_FIRST)
+	return sysNetValues[idx].String(), nil
 }
 
-type SocketDomainArgument uint64
```

Comment:
What about socket domain type? No need for this?

Reply:
I'm abolishing the different types since all of them have the same behaviour of String and Value. I'm focusing on declaring flags (based on origin - include files) and declaring slices like `socketDomainValues = []SystemFunctionArgument{` which hold a instance copy of the related flags already made public. For the particular case of `SocketDomain`, `socketDomainValues` store the values and they are parsed by `ParseSocketDomainArgument()`.

You can better understand the idea looking the `AT_*` declarations and the "`faccessatFlagsValues`, `fchmodatFlagsValues` and `execveAtFlagsValues`" holders which make use of only a fragment of those flags based on the event to be parsed. Before we were iterating the whole bunch for events that don't use those flags.